crates/ruff_linter/src/rules/ruff/rules/split_of_static_string.rs

```diff
@@ -0,0 +1,166 @@
+use ruff_diagnostics::{Diagnostic, Edit, Fix, FixAvailability, Violation};
+use ruff_macros::{derive_message_formats, violation};
+use ruff_python_ast::{
+    Expr, ExprCall, ExprContext, ExprList, ExprStringLiteral, StringLiteral, StringLiteralFlags,
+    StringLiteralValue,
+};
+use ruff_text_size::{Ranged, TextRange};
+
+use crate::checkers::ast::Checker;
+
+/// ## What it does
+/// Checks for str.split calls that can be replaced with a `list` literal.
+///
+/// ## Why is this bad?
+/// List literals are more readable and do not require the overhead of calling `str.split`.
+///
+/// ## Example
+/// ```python
+/// "a,b,c,d".split(",")
+/// ```
+///
+/// Use instead:
+/// ```python
+/// ["a", "b", "c", "d"]
+/// ```
+///
+/// ## References
+///
+/// - [Python documentation: `str.split`](https://docs.python.org/3/library/stdtypes.html#str.split)
+///
+/// ```
+#[violation]
+pub struct SplitOfStaticString;
+
+impl Violation for SplitOfStaticString {
+    const FIX_AVAILABILITY: FixAvailability = FixAvailability::Sometimes;
+
+    #[derive_message_formats]
+    fn message(&self) -> String {
+        format!("Consider using a list instead of string split")
+    }
+
+    fn fix_title(&self) -> Option<String> {
+        Some(format!("Replace string split with list literal"))
+    }
+}
+
+fn construct_replacement(list_items: &[&str]) -> Expr {
+    Expr::List(ExprList {
+        elts: list_items
+            .iter()
+            .map(|list_item| {
+                Expr::StringLiteral(ExprStringLiteral {
+                    value: StringLiteralValue::single(StringLiteral {
+                        value: (*list_item).to_string().into_boxed_str(),
+                        range: TextRange::default(),
+                        flags: StringLiteralFlags::default(),
+                    }),
+                    range: TextRange::default(),
+                })
+            })
+            .collect(),
+        ctx: ExprContext::Load,
+        range: TextRange::default(),
+    })
+}
+
+fn split_default(str_value: &str, max_split: usize) -> Option<Expr> {
+    // From the Python documentation:
+    // > If sep is not specified or is None, a different splitting algorithm is applied: runs of
+    // > consecutive whitespace are regarded as a single separator, and the result will contain
+    // > no empty strings at the start or end if the string has leading or trailing whitespace.
+    // > Consequently, splitting an empty string or a string consisting of just whitespace with
+    // > a None separator returns [].
+    // https://docs.python.org/3/library/stdtypes.html#str.split
+    if max_split == 0 {
+        let list_items: Vec<&str> = str_value.split_whitespace().collect();
+        Some(construct_replacement(&list_items))
+    } else {
+        // Autofix for maxsplit without separator not yet implemented
+        None
```

Comment:
What makes `maxsplit` with the default separator difficult to implement?

Reply:
There is no equivalent to `split` for `split_whitespace`. Implementing it is not that much work, but it would be simplified greatly when this experimental API is stabilised (https://doc.rust-lang.org/std/str/struct.SplitWhitespace.html#method.remainder). I don't think it's worth to implement the logic now. I'll add a comment.